wemake_python_styleguide/visitors/tokenize/comments.py

```diff
@@ -183,3 +185,90 @@ def _check_empty_line_after_codding(
             if next_token.exact_type not in self._newlines:  # pragma: no cover
                 self.add_violation(EmptyLineAfterCodingViolation(token))
             break
+
+
+@final
+class ShebangVisitor(BaseTokenVisitor):
+    """Checks the first shebang in the file."""
+
+    def visit(self, token: tokenize.TokenInfo) -> None:
+        """
+        Checks if there is an executable mismatch.
+
+        Raises:
+            ExecutableMismatchViolation
+
+        """
+        is_first_token = (token == self.file_tokens[0])
+
+        if not is_first_token:
+            return
+
+        shebang_token = self._get_shebang_token()
+
+        self._check_executable_mismatch(shebang_token)
+        if shebang_token is not None:
+            self._check_valid_shebang(shebang_token)
+
+    def _check_valid_shebang_line(self, line) -> bool:
+        return re.match(r'(\s*)#!', line) is not None
+
+    def _get_shebang_token(self) -> Optional[tokenize.TokenInfo]:
+        all_tokens = iter(self.file_tokens)
+
+        current_token = next(all_tokens)
+
+        while current_token.exact_type in NEWLINES.union({tokenize.COMMENT}):
```

Comment:
Let's make move `.union` operation into a variable. Maybe even class-level one. What do you think?